src/cloudant/document.py

```diff
@@ -96,6 +98,77 @@ def save(self):
         put_resp.raise_for_status()
         return
 
+    # Update Actions
+    # These are handy functions to use with update_field below.
+    @staticmethod
+    def field_append(doc, field, value):
+        """Append a value to a field in a doc."""
+        doc[field].append(value)
+    @staticmethod
+    def field_remove(doc, field, value):
+        """Remove a value from a field in a doc."""
+        doc[field].remove(value)
+    @staticmethod
+    def field_replace(doc, field, value):
+        """Replace a field in a doc with a value."""
+        doc[field] = value
+
+    def _update_field(self, action, field, value, max_tries, tries=0):
+        """
+        Private update_field method. Wrapped by CloudantDocument.update.
+        Tracks a "tries" var to help limit recursion.
+
+        """
+        # Refresh our view of the document.
+        self.fetch()
+
+        # Update the field.
+        action(self, field, value)
+
+        # Attempt to save, retrying conflicts up to max_tries.
+        try:
+            self.save()
+        except requests.HTTPError as ex:
+            if tries < max_tries and ex.response.status_code == 409:
+                return self._update_field(
+                    action, field, value, max_tries, tries=tries+1)
+            raise
```

Comment:
If this does fail out, there are going to be `max_tries` entries on the stack trace. Not sure if this is desired or not.